demos/code/Blocks/Chapter 5 - Functions.js

```diff
@@ -370,249 +248,210 @@ Blockly.Blocks['user_function'] = {
 				
 				break;
 			}
-			
+
 			ptr = ptr.parentBlock_;
 		}
-		
+
 		//Check if a function has been redeclared
-		
+
 		//Check function parameter blocks
-		ptr = this.getInputTargetBlock('valinp1');
-		
+		ptr = this.getInputTargetBlock('valueInput');
+
 		while(ptr){
-			
-			if(ptr.type !== "func_parameters"){
+
+			if(ptr.type !== "function_parameters" && ptr.type !== "pointer_operator"){
 				TT += 'Error, only the function parameter block is allowed in the function parameter.\n';
 				break;
 			}
-			
+
 			ptr = ptr.childBlocks_[0];
 		}
-		
+
 		//Function parameter block end
-		
+
 		//Check string Library
-		
-		if(this.typeName_ == "string"){
-				
+
+		if(this.typeName_ === "string"){
+
 			//create an instance of C_Include
 			var librarySearch = C_Include;
-			
+
 			var libFound = librarySearch.search_library(this, ['include_string']);
-			
+
 			if(!libFound){
 				TT += "Error, <string> library must be included.\n";
 			}
 		}
-		
+
 		//Check if a function requires a return
-		
-		
-		
+
+
+
 		if(TT.length > 0){
 			this.setWarningText(TT);
 		}
 		else {
 			this.setWarningText(null);
 		}
-	},
-};
-
-Blockly.C['user_function'] = function(block) {
-	var stateinp1 = Blockly.C.statementToCode(block, 'stateinp1');
-	
-	var val1 = Blockly.C.valueToCode(block, 'valinp1', Blockly.C.ORDER_ATOMIC);
-
-	C = C_Include;
-	
-	// TODO: Assemble C into code variable.
-	var code = '';
-	var std = '';
-	
-	if(block.funcProp_[1] === 'string' && !C.using.std(block)){
-		std += 'std::';
-	}
-	
-	//If it's not a constructor
-	if(!block.isConstructor_){
-		if(block.funcProp_[0]){
-			code += block.getField('const').getText() + ' ';
-		}
-		
-		code += std + block.funcProp_[1] + block.funcProp_[2] + ' ';
 	}
 
-	//Simple name formatting
-	code += block.funcProp_[3] + '(';
+};
+Blockly.C['function_declaration'] = function(block) {
```

Comment:
This is just a rename of the block. The reference in index.html has been updated as well.